Sources/WalletConnectSign/Engine/Common/SessionEngine.swift

```diff
@@ -59,11 +59,16 @@ final class SessionEngine {
         try await networkingInteractor.request(rpcRequest, topic: request.topic, protocolMethod: SessionRequestProtocolMethod())
     }
 
-    func respondSessionRequest(topic: String, requestId: RPCID, response: RPCResult) async throws {
+    func respondSessionRequest(topic: String, requestId: RPCID, result: RPCResult) async throws {
         guard sessionStore.hasSession(forTopic: topic) else {
             throw Errors.sessionNotFound(topic: topic)
         }
-        let response = RPCResponse(id: requestId, result: response)
+      
+      var response = RPCResponse(id: requestId, result: result)
+      
+      if case .error(let value) = result {
+        response = RPCResponse(id: requestId, error: value);
+      }
```

Comment:
hey @kirill09 that's a good point, the initialiser is incorrect for an error response but after your changes tests are not passing. 

Did you encounter any problems with current implementation?

Reply:
hi @llbartekll In the current implementation, I was getting an RPCResponse result object and it had an error in it. 
Which was causing the wrong behavior. 
Actions didn't get to the point of running tests. I think these errors are not related to my changes. Can you restart the actions?

Reply:
it's interesting because the response actually decodes to error. take look at this test, it passes:
https://github.com/WalletConnect/WalletConnectSwiftV2/blob/84197cd011da4b1986fc7a431cdf3f0f32199db1/Example/IntegrationTests/Sign/SignClientTests.swift#L247
and after your changes it does not, i will take a closer look at it

Reply:
@llbartekll Perhaps the latest commit will fix the tests